### PR TITLE
Initial PR/commit. Added dependencies.

### PR DESCRIPTION
**Pull request by catslanderer**

### CODE REVIEW

File: data/models/auth-model.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    return 'taken';
+  } else {
+    return 'available';
+  }
+};
+
+const registerUser = async user => {
+  const password = bcrypt.hashSync(user.password, 8);
+  return db('users').insert({
+    username: user.username.toLowerCase(),
```
Comment (DaftBeowulf) [inline data/models/auth-model.js:19]: I didn't comment on this in the week's earlier projects, but while this isn't a terrible idea when validating the data (if you want the usernames to basically be case sensitive when signing on), is it entirely necessary to set it all to lowercase when inserting into the database? you'd have no way then of ever checking exactly how the user submitted their data when registering, then, and they might not be expecting a normalized version of their username if was ever reflected anywhere in the app's front end

File: data/models/auth-model.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    password,
+    department: user.department
```
Comment (DaftBeowulf) [inline data/models/auth-model.js:21]: make it even more fun -- destructure all the properties off of the passed-in user and do the equivalent property-value syntax for all three properties

**Review by DaftBeowulf — COMMENTED** [review]
Great job Chris, you're getting very skilled at separating middleware and unique-purpose functions/configs into their own file structure -- great work transitioning into JWTs

File: data/models/auth-model.js (hunk 1 above)
Comment (catslanderer) [reply data/models/auth-model.js:19]: I was trying to follow Google's best practices, but I see that the user would not expect "Hello, chris" if they entered their username as "Chris". I think I can fix this.

**Pull request merged**